ext/intl/common/common_date.cpp

```diff
@@ -109,35 +108,35 @@ U_CFUNC int intl_datetime_decompose(zval *z, double *millis, TimeZone **tz,
 		*millis = ZEND_NAN;
 	}
 	if (tz) {
-		*tz = NULL;
+		*tz = nullptr;
 	}
 
 	if (millis) {
-		php_date_obj *datetime;
-
-		ZVAL_STRING(&zfuncname, "getTimestamp");
-		if (call_user_function(NULL, z, &zfuncname, &retval, 0, NULL)
-				!= SUCCESS || Z_TYPE(retval) != IS_LONG) {
-			spprintf(&message, 0, "%s: error calling ::getTimeStamp() on the "
-					"object", func);
-			intl_errors_set(err, U_INTERNAL_PROGRAM_ERROR,
-				message, 1);
+		auto *getTimestampMethod = static_cast<zend_function *>(zend_hash_str_find_ptr(&obj->ce->function_table, ZEND_STRL("gettimestamp")));
```

Comment:
nit: `auto getTimestampMethod`, it is deduced to zend_function without needing the `*`